src/solver/conv_mlir_igemm_fwd_xdlops.cpp

```diff
@@ -139,14 +139,33 @@ bool PerformanceConvMlirIgemmXdlops::SetNextValue(const ConvolutionContext& /*co
             break;
         if(!NextTwoPower<16, 256>(GemmNPerBlock))
             break;
-        if(!NextTwoPower<1, 8>(GemmKPerBlock))
+        if(!NextTwoPower<4, 8>(GemmKPerBlock))
             break;
         if(!NextTwoPower<4, 128>(GemmMPerWave))
             break;
         if(!NextTwoPower<16, 128>(GemmNPerWave))
             break;
-        if(!NextTwoPower<4, 8>(GemmKPACKSize))
-            break;
+        if(config.IsFp32())
+        {
+            if(!NextTwoPower<1, 4>(GemmKPACKSize))
+                break;
+        }
+        else if(config.IsFp16())
+        {
+            if(!NextTwoPower<1, 8>(GemmKPACKSize))
+            {
+                if(GemmKPACKSize == 2)
+                {
+                    GemmKPACKSize = 4;
+                }
+                break;
+            }
+        }
+        else
+        {
+            if(!NextTwoPower<1, 1>(GemmKPACKSize))
+                break;
+        }
```

Comment:
[Recommendation] To get rid of ambiguity
```suggestion
        ; // Nop. Leave GemmKPACKSize equal to 1.
```

Reply:
Changed in the very recent commit.